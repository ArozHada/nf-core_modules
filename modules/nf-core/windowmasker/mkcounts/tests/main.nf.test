nextflow_process {

    name "Test Process WINDOWMASKER_MKCOUNTS"
    script "../main.nf"
    process "WINDOWMASKER_MKCOUNTS"

    tag "modules"
    tag "modules_nfcore"
    tag "windowmasker"
    tag "windowmasker/mkcounts"

    test("Should run without failures") {

        when {
            params {
                // define parameters here. Example:
                // outdir = "tests/results"
            }
            process {
                """
                input[0] = [
                    [id: "test" ],
                    [ file(params.test_data['sarscov2']['genome']['genome_fasta'], checkIfExists: true)]
                ]
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }
    test("sarscov2 - fasta - stub") {

        options '-stub'

        when {
            process {
                """
                input[0] = [
                    [id: "test" ],
                    [ file(params.test_data['sarscov2']['genome']['genome_fasta'], checkIfExists: true)]
                ]
                """
            }
        }
        then {
            assert process.success
            assert snapshot(process.out.versions).match()
        }
    }

}
