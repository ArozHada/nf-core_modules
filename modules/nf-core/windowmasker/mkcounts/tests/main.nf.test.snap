{
    "sarscov2 - fasta - stub": {
        "content": [
            [
                "versions.yml:md5,57ae356f69298e25eb5d070900865cf2"
            ]
        ],
        "timestamp": "2024-02-15T12:45:33.501196"
    },
    "Should run without failures": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.txt:md5,5f5d7e926fdf13b0c57651f962cc1253"
                    ]
                ],
                "1": [
                    "versions.yml:md5,57ae356f69298e25eb5d070900865cf2"
                ],
                "counts": [
                    [
                        {
                            "id": "test"
                        },
                        "test.txt:md5,5f5d7e926fdf13b0c57651f962cc1253"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,57ae356f69298e25eb5d070900865cf2"
                ]
            }
        ],
        "timestamp": "2024-02-15T12:40:56.965802"
    }
}